R/util-MakeWorkflowList.R

```diff
@@ -43,46 +43,68 @@ MakeWorkflowList <- function(
     full.names = TRUE,
     recursive = bRecursive
   )
+  names(yaml_files) <- list.files(
+    path,
+    pattern = "\\.yaml$",
+    full.names = FALSE,
+    recursive = bRecursive
+  )
+
+  # if `strNames` is not null, subset the workflow list to only include
+  # files that match the character vector (`strNames`)
+  good_strNames <- c(
+    strNames,
+    paste(strNames, ".yaml", sep = "")
+  )
+
+  if (!is.null(strNames)) {
+    if (bExact) {
+      yaml_files <- purrr::keep(yaml_files, names(yaml_files) %in% good_strNames)
+    } else {
+      yaml_files <- purrr::keep(yaml_files, grepl(paste(strNames, collapse = "|"), names(yaml_files))) # this may have unintended consequences, AE vs DATAENT
+    }
+  }
 
-  workflows <- yaml_files %>%
-    purrr::map(function(yaml_file) {
+  workflows <-  purrr::map2(
+    yaml_files,
+    names(yaml_files),
+    function(yaml_file, file_name) {
       # read the individual YAML file
       workflow <- yaml::read_yaml(yaml_file)
 
       # set the `path` for logging purposes
       workflow$path <- yaml_file
 
       # each workflow should have an $meta and $steps $meta$ID attributes
-      stopifnot(utils::hasName(workflow, "meta"))
-      stopifnot(utils::hasName(workflow, "steps"))
-      stopifnot(utils::hasName(workflow$meta, "ID"))
+      if(!utils::hasName(workflow, "meta")){
+        cli::cli_abort(c("{file_name} must contain `meta` attributes."))
+      }
+      if(!utils::hasName(workflow, "steps")){
+        cli::cli_abort(c("{file_name} must contain `steps` attributes."))
+      }
+      if(!utils::hasName(workflow$meta, "ID")){
```

Comment:
```suggestion
      if(!utils::hasName(workflow$meta, "Type")){
        cli::cli_abort(c("{file_name} must contain `Type` attribute in `meta` section."))
      }
      if(!utils::hasName(workflow$meta, "ID")){
```

Reply:
`Type` will be relied on in the data model I/O so would be nice to require here.